### PR TITLE
Use 32 bit datatypes in tests

### PR DESCRIPTION
**Pull request by berland**
**Issue**
Resolves hundreds of warnings from the test suite

**Approach**
Use special function from hypothesis to ensure correct dtypes.

## Pre review checklist

- [x] Added appropriate release note label
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [ ] Updated documentation
- [ ] Ensured new behaviour is tested

Adding labels helps the maintainers when writing release notes. This is the [list of release note labels](https://github.com/equinor/ert/labels?q=release-notes).

### CODE REVIEW

File: tests/test_config_parsing/summary_generator.py [hunk 1]
```diff
@@ -231,15 +232,21 @@ class Unsmry:
...
     def to_file(
         self, filelike, file_format: ecl_data_io.Format = ecl_data_io.Format.UNFORMATTED
     ):
         ecl_data_io.write(filelike, self.to_ecl(), file_format)
 
 
-positive_floats = st.floats(min_value=0.1, allow_nan=False, allow_infinity=False)
+positive_floats = from_dtype(
+    np.dtype(np.float32),
+    min_value=0.10000000149011612,  # 0.1 rounded to float32
```
Comment (xjules) [inline tests/test_config_parsing/summary_generator.py:246]: Hm, why not `min_value=0.1`?

File: tests/performance_tests/performance_utils.py [hunk 2]
```diff
@@ -17,14 +17,14 @@
 
 def write_summary_spec(file, keywords):
     content = [
-        ("INTEHEAD", [1, 100]),
+        ("INTEHEAD", array([1, 100], dtype=numpy.int32)),
```
Comment (xjules) [inline tests/performance_tests/performance_utils.py:20]: btw, you can also do `array(...).astype(np.int32)`, which applies for all the others.

Comment (berland) [reply tests/performance_tests/performance_utils.py:20]: are you sure that will not create the array with double values and then do the astype-conversion?

Comment (xjules) [reply tests/performance_tests/performance_utils.py:20]: Good point!

File: tests/test_config_parsing/summary_generator.py (hunk 1 above)
Comment (xjules) [reply tests/test_config_parsing/summary_generator.py:246]: Can you put this line as a comment there ? Just that people / us would understand when we look at it in a year or so :smile:

Comment (berland) [reply tests/test_config_parsing/summary_generator.py:246]: Comment updated! :)

Comment (berland) [reply tests/test_config_parsing/summary_generator.py]: No, changed to this instead:
`
+    min_value=np.float32(0.1),
`

**Review by xjules — APPROVED** [review]
Very good! :rocket:

**Pull request merged**